code/game/dna/dna_modifier.dm

```diff
@@ -93,6 +93,18 @@
 	add_fingerprint(usr)
 	return
 
+/obj/machinery/dna_scannernew/crowbarDestroy(mob/user)
+	if(occupant)
+		user << "<span class='warning'>You cannot disassemble \the [src], it's occupado.</span>"
+		return
+	return ..()
+
+/obj/machinery/dna_scannernew/Destroy()
+
+	go_out() //Eject everything
```

Comment:
There's no reason to delete the mob ever. In the case of explosions, it should manually call ex_act() on the occupant unless explosion resistance is desired